### PR TITLE
Reading Queries from Config yaml

### LINKED ISSUE #3: Making Queries Configurable
Right Now, The queries are present in the [queries.go](https://github.com/deislabs/smi-metrics/blob/master/pkg/metrics/queries.go) file and are specific to linkerd. 

To make this repo support more implementations, we need a way to make the prom quieries pluggable. This can be solved by having each implementation have a ConfigMap with the queries.

The corresponding ConfigMap is installed and dynamically loaded as a configuration field into Viper during initialisation.

@grampelberg

### PR DESCRIPTION
**Pull request by Pothulapati**
Fixes #3 

This PR makes the queries, read from the Config File. So, as to support more meshes that have different  queries.

@grampelberg

### CODE REVIEW

File: pkg/metrics/handler.go [hunk 1]
```diff
@@ -14,19 +14,25 @@ import (
 type Handler struct {
 	client promv1.API
 
+	resourceQueries map[string]string
```
Comment (grampelberg) [inline pkg/metrics/handler.go]: I would prefer that we have a `queries` struct that contains the different types.

File: linkerd.yaml [hunk 2]
```diff
@@ -0,0 +1,157 @@
+config:
+  prometheus-url: http://linkerd-prometheus.linkerd.svc.cluster.local:9090
+  resourceQueries:
+    p99_response_latency : 'histogram_quantile(
```
[the remark this answers is not in the document]
Comment (Pothulapati) [reply linkerd.yaml]: Done

File: Tiltfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-k8s_yaml(local("helm template chart -f dev.yaml --name dev"))
+k8s_yaml(local("helm template chart -f linkerd.yaml --name dev"))
```
[the remark this answers is not in the document]
Comment (Pothulapati) [reply Tiltfile]: Updated

**Comment by Pothulapati** [comment]
Fixed @grampelberg

**Comment by Pothulapati** [comment]
Oops, Didn't recognise the linting issues from `golangci-lint`, Now updated the PR to pass it. 
Expect a PR with circleci integration next :)

**Pull request merged**